src/OpenDiffix.Core/Analyzer.fs

```diff
@@ -403,11 +403,12 @@ let private computeNoiseLayers anonParams query =
   let sqlSeed =
     query
     |> collectGroupingExpressions
-    |> Seq.map normalizeNoiseLayerExpression
-    |> Seq.collect (collectSeedMaterials rangeColumns)
-    |> String.join ","
-    |> System.Text.Encoding.UTF8.GetBytes
-    |> Hash.bytes
+    |> Seq.map (
+      normalizeNoiseLayerExpression
+      >> collectSeedMaterials rangeColumns
+      >> String.join ","
+    )
+    |> Hash.strings 0UL
```

Comment:
Could we end up in a situation where bucket labels cancel each other out because of the XOR?

Reply:
I don't think this is a problem if it happens by chance for different expressions, since 0 is a seed value just as valid as any other. But it does happen when the same expression is used multiple times for grouping, cancelling its effect on the bucket seed. I will normalize this case in a future PR. Good catch!